cli/create_test.go

```diff
@@ -663,6 +666,556 @@ func TestCreateWithRichParameters(t *testing.T) {
+		member, _ := coderdtest.CreateAnotherUser(t, client, owner.OrganizationID)
+
+		// Given: a template version with a preset that defines one parameter
+		preset := proto.Preset{
+			Name: "preset-test",
+			Parameters: []*proto.PresetParameter{
+				{Name: firstParameterName, Value: secondOptionalParameterValue},
+			},
+		}
+		version := coderdtest.CreateTemplateVersion(t, client, owner.OrganizationID, echoResponses(&preset))
+		coderdtest.AwaitTemplateVersionJobCompleted(t, client, version.ID)
+		template := coderdtest.CreateTemplate(t, client, owner.OrganizationID, version.ID)
+
+		// When: running the create command with the specified preset
+		workspaceName := "my-workspace"
+		inv, root := clitest.New(t, "create", workspaceName, "--template", template.Name, "--preset", preset.Name)
+		clitest.SetupConfig(t, member, root)
+		doneChan := make(chan struct{})
+		pty := ptytest.New(t).Attach(inv)
+		go func() {
+			defer close(doneChan)
+			err := inv.Run()
+			assert.NoError(t, err)
+		}()
+
+		// Should: display the selected preset as well as its parameters
+		presetName := fmt.Sprintf("Preset '%s' applied:", preset.Name)
+		pty.ExpectMatch(presetName)
+		pty.ExpectMatch(fmt.Sprintf("%s: '%s'", firstParameterName, secondOptionalParameterValue))
+
+		// Should: prompt for the missing parameter
+		pty.ExpectMatch(thirdParameterDescription)
+		pty.WriteLine(thirdParameterValue)
+		pty.ExpectMatch("Confirm create?")
+		pty.WriteLine("yes")
+
+		<-doneChan
+
+		// Verify if the new workspace uses expected parameters.
+		ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitLong)
+		defer cancel()
+
+		tvPresets, err := client.TemplateVersionPresets(ctx, version.ID)
+		require.NoError(t, err)
+		require.Len(t, tvPresets, 1)
+
+		workspaces, err := client.Workspaces(ctx, codersdk.WorkspaceFilter{
+			Name: workspaceName,
+		})
+		require.NoError(t, err)
+		require.Len(t, workspaces.Workspaces, 1)
+
+		// Should: create a workspace using the expected template version and the preset-defined parameters
+		workspaceLatestBuild := workspaces.Workspaces[0].LatestBuild
+		require.Equal(t, version.ID, workspaceLatestBuild.TemplateVersionID)
+		require.Equal(t, tvPresets[0].ID, *workspaceLatestBuild.TemplateVersionPresetID)
+		buildParameters, err := client.WorkspaceBuildParameters(ctx, workspaceLatestBuild.ID)
+		require.NoError(t, err)
+		require.Len(t, buildParameters, 2)
+		require.Contains(t, buildParameters, codersdk.WorkspaceBuildParameter{Name: firstParameterName, Value: secondOptionalParameterValue})
+		require.Contains(t, buildParameters, codersdk.WorkspaceBuildParameter{Name: thirdParameterName, Value: thirdParameterValue})
+	})
+
+	// This test verifies that when a template has a default preset,
+	// and the user does not provide the `--preset` flag,
+	// the CLI prompts the user to select a preset, and the default preset is listed first.
+	t.Run("PromptsUserToSelectPresetWhenNotSpecified", func(t *testing.T) {
```

Comment:
Could we also have a test that validates that the default preset is selected when skipping the prompt via `--yes`?

Reply:
@johnstcn after discussing internally with @bartekgatzcoder, we decided not to automatically use the default preset.
Instead, if no `--preset` flag is passed and the selected template has presets, the CLI will always prompt the user to select a preset, ensuring they are aware of the choice being made. To improve the experience, we make sure the default preset appears first in the list.

This approach aligns with the behavior in the UI for consistency.

Reply:
Were you and @bartekgatzcoder both aware and in consideration of the `--yes` flag? It occurs to me that the point of the flag might be to allow the user to force non-interactive use.

Reply:
Currently (without this implementation), even if you pass the `--yes` flag, the `create` command still prompts you to select a template and the parameters if they’re not already specified via flags, so prompting for a preset would follow the same logic and keep the experience consistent.

Reply:
There was a misunderstanding on my end. After syncing with @bartekgatzcoder  again, it’s clear that if a template has a default preset and the user doesn’t explicitly specify one, the CLI should automatically use the default. The user should only be prompted to select a preset, if the template has presets but no default is defined. Addressed in https://github.com/coder/coder/pull/18912/commits/dbe0685ac9819cfb4b281cdff2a236ae1f95b4ee